pkgs/by-name/op/opentelemetry-cpp/package.nix

```diff
@@ -53,17 +67,22 @@ stdenv.mkDerivation (finalAttrs: {
 
   strictDeps = true;
 
-  cmakeFlags = [
-    "-DBUILD_SHARED_LIBS=ON"
-    "-DWITH_OTLP_HTTP=ON"
-    "-DWITH_OTLP_GRPC=ON"
-    "-DWITH_ABSEIL=ON"
-    "-DWITH_PROMETHEUS=ON"
-    "-DWITH_ELASTICSEARCH=ON"
-    "-DWITH_ZIPKIN=ON"
-    "-DWITH_BENCHMARK=OFF"
-    "-DOTELCPP_PROTO_PATH=${opentelemetry-proto}"
-  ];
+  cmakeFlags =
+    [
+      (lib.cmakeBool "BUILD_SHARED_LIBS" (!stdenv.hostPlatform.isStatic))
+      (lib.cmakeBool "WITH_ABSEIL" true)
```

Comment:
`WITH_ABSEIL` is removed in the commit after the 1.20.0 release. So probably abseil-cpp and the configure flag should be removed from here, too, since they are not future proof. See https://github.com/open-telemetry/opentelemetry-cpp/issues/3116